src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -147,7 +187,61 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query query = null;
+        Preconditions.checkNotNull(this.fieldName, "Fieldname of a TermQuery cannot be null.");
+        Preconditions.checkNotNull(this.value, "Value of a TermQuery cannot be null.");
```

Comment:
+1 to this, I think this is a great ide, for now we can do the validate method and figure out where to hook it in at a later time.

Also as an aside, I'd love to get rid of `Preconditions.checkNotNull` and just use `Objects.requireNonNull` since it's part of the JVM, but that's just personal preference.